### ISSUE TITLE
Null pointer exception on creating child node for parallel method execution

### ISSUE THREAD

**Issue by mirroredthoughts** [anshooarora/extentreports-java#754]
Hi ,

NPE is thrown on executing tests in parallel at method level while creating child node for the multiple @Test methods in same test class. Able to create child node only for one @Test method. Remaining ones are getting NPE.

Please find below testng listner class and test class.

beforeInvocation() creates child node and appends to the parent. But parentTest.get() is given Null for the all test methods excepts first test method in test class. 

I have used the example mentioned in extent Reports java docs for v3.0.1.
[NPE-ParallelMethods.zip](https://github.com/anshooarora/extentreports-java/files/732603/NPE-ParallelMethods.zip)

**Comment by email2vimalraj** [comment]
Few things to be noted here:

- You are flushing the extent instance after every method. Move the `extent.flush()` to the `onFinish(ISuite suite)`
- Instead of using `ThreadLocal`, use `InheritableThreadLocal`. More about this [When to use InheritableThreadLocal?](http://geekexplains.blogspot.com/2009/02/threadlocal-inheritablethreadlocal-in.html)

In case you want some out of the box solution, I would recommend you to try the following TestNG listener library for the Extent Report: [TestNGExtentsReport](https://github.com/email2vimalraj/TestNGExtentsReport)

**Comment by payalgandhi** [comment]
Hi, I am running parallel testing using TestNG and extent report. I am facing issue with ExtentTest.set() in few testcases it is passing and in few testcases it is failing after this line ExtentTest.set() giving me NullPointerException error can someone help?

**Comment by anshooarora** [comment]
Please open a new issue explaining what you are doing, along with code snippets that may help us understand this better.
